src/helpers/in-memory-cache.ts

```diff
@@ -0,0 +1,48 @@
+import type { VirtualCurrencies } from "../entities/virtual-currencies";
+
+interface CacheEntry<T> {
+  data: T;
+  timestamp: number;
+}
+
+export class InMemoryCache {
+  private virtualCurrenciesCache = new Map<
```

Comment:
Hmm would it make sense to make this a truly generic class instead of having maps for each piece of data we might want to cache? That way we would have different caches for `VirtualCurrencies` and other things.

Reply:
I considered this! There were two reasons I went with having one cache that could be expanded to include caching other items in the future:
- That's the pattern used by our DeviceCache in `purchases-ios`/`purchases-android`
- If we had one `InMemoryCache` instance per item that we're caching, that could lead to us storing lots of `InMemoryCache` objects in the `Purchases` object. That's not necessarily bad, but could lead to bloating the Purchases class a bit with a lot of cache objects, instead of just having one cache object.